docs/rules/forbid-elements.md

```diff
@@ -0,0 +1,39 @@
+# Forbid certain elements (forbid-elements)
+
+You may want to forbid usage of certain elements in favor of others, (e.g. forbid all `<div />` and use `<Box />` instead). This rule allows you to configure a list of forbidden elements and to specify their desired replacements.
+
+## Rule Details
+
+This rule checks all JSX components and verifies that no forbidden elements are used. When it detects a forbidden element, the error message will contain the specified elements that you should use instead. This rule is off by default. If on, no elements are forbidden by default.
```

Comment:
If we consider `React.createElement` calls that can be mapped back to JSX as "valid", then  `React.createElement('Modal')`, and `React.createElement(button)` would not be valid.

`<Modal />` translates to `React.createClass(Modal)`, and there's no way I know that we can get it to translate to `React.createClass('Modal')`.

That's why I think this rule should not bother to make checks that cannot be mapped back to JSX, and can be left to a different rule.